src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentHighlighting/DocumentHighlightEndpoint.cs

```diff
@@ -30,10 +30,11 @@ internal class DocumentHighlightEndpoint : AbstractRazorDelegatingEndpoint<Docum
         public RegistrationExtensionResult GetRegistration(VSInternalClientCapabilities clientCapabilities)
         {
             const string ServerCapability = "documentHighlightProvider";
-            var options = new DocumentHighlightOptions
-            {
-                WorkDoneProgress = false
-            };
+            var options = new SumType<bool, DocumentHighlightOptions>(
```

Comment:
Then the capability accepts a SumType our property assigner will fail if we don't do the cast beforehand.